oak_ml_transparency/mnist/eval.py

```diff
@@ -9,49 +29,88 @@
+
+    # Load some dataset to use as input to the model and for generating adversarial examples.
+    (train_images, train_labels), (test_images, test_labels) = tf.keras.datasets.mnist.load_data()
+    test_images = test_images[:10].reshape(-1, 28 * 28) / 255.0
+
+    print("Input shape: ", test_images.shape)
+    out = mnist_model.predict(test_images)
+
+    # Generate adversarial examples from the input data, and get predictions for them. 
+    # Based on https://github.com/cleverhans-lab/cleverhans/blob/master/tutorials/jax/mnist_tutorial.py
+
+    y_pred_fgm = []
+    y_pred_pgd = []
+
+    for i in range(10):
+        x = test_images[i:i+1]
+        y = test_labels[i:i+1]
+        x_fgm = fast_gradient_method(mnist_model, x, EPS, np.inf)
+        y_pred_fgm.append(mnist_model(x_fgm).numpy()[0])
 
-mnist_path = sys.argv[1]
+        x_pgd = projected_gradient_descent(mnist_model, x, EPS, 0.01, 40, np.inf)
+        y_pred_pgd.append(mnist_model(x_pgd).numpy()[0])
 
-# Load the mnist model
-mnist_model = keras.models.load_model(mnist_path)
-print("Successfully loaded the model from", mnist_path)
+    # Compute accuracy
+    metric = tf.keras.metrics.SparseCategoricalAccuracy()
+    metric.update_state(test_labels[:10], out)
+    test_acc = metric.result().numpy() * 100
 
-# Load some dataset to use as input to the model and for generating adversarial examples.
-(train_images, train_labels), (test_images, test_labels) = tf.keras.datasets.mnist.load_data()
-test_images = test_images[:10].reshape(-1, 28 * 28) / 255.0
+    metric.reset_state()
+    metric.update_state(test_labels[:10], y_pred_fgm)
+    fgm_acc = metric.result().numpy() * 100
 
-print("Input shape: ", test_images.shape)
-out = mnist_model.predict(test_images)
+    metric.reset_state()
+    metric.update_state(test_labels[:10], y_pred_pgd)
+    pgd_acc = metric.result().numpy() * 100
+    
+    result = {
+        "test_acc": test_acc,
+        "fgm_acc": fgm_acc,
+        "pgd_acc": pgd_acc
+    }
+    return result
 
-# Generate adversarial examples from the input data, and get predictions for them. 
-# Based on https://github.com/cleverhans-lab/cleverhans/blob/master/tutorials/jax/mnist_tutorial.py
+def cleanup_archive(file: tarfile.TarFile):
+    for member in file.getmembers():
+        if os.path.isdir(member.name):
+            shutil.rmtree(member.name, ignore_errors=True)
+        elif os.path.exists(member.name):
+            os.remove(member.name)
+    file.close()
 
-y_pred_fgm = []
-y_pred_pgd = []
+def main():
+    parser = argparse.ArgumentParser(
+                prog='MNIST evaluation',
+                description='Evaluates an MNIST model against adversarial examples')
 
-for i in range(10):
-    x = test_images[i:i+1]
-    y = test_labels[i:i+1]
-    x_fgm = fast_gradient_method(mnist_model, x, EPS, np.inf)
-    y_pred_fgm.append(mnist_model(x_fgm).numpy()[0])
+    parser.add_argument('-m', '--model', help="the model as a compressed tar archive")
```

Comment:
Personally I would avoid short flags at all, but I am not sure if that's possible with this python flag library